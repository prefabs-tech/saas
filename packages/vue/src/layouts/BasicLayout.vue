<template>
  <OriginalBasicLayout :no-locale-switcher="noLocaleSwitcher">
    <template #header>
      <AppHeader>
        <template #logo>
          <Logo :route="home" />
        </template>
        <template #addon>
          <slot name="addon" />
        </template>
        <template #menu>
          <MainMenu class="main-menu" :routes="routes" />
        </template>
        <template #userMenu>
          <UserMenu v-if="showUserMenu" />
        </template>
      </AppHeader>
    </template>

    <slot></slot>
  </OriginalBasicLayout>
</template>

<script lang="ts">
export default {
  name: "BasicLayout",
};
</script>

<script setup lang="ts">
import { useConfig } from "@dzangolab/vue3-config";
import {
  AppHeader,
  BasicLayout as OriginalBasicLayout,
  Logo,
  MainMenu,
} from "@dzangolab/vue3-layout";
import { storeToRefs } from "pinia";
import { computed } from "vue";
import { useRouter } from "vue-router";

import UserMenu from "../components/UserMenu.vue";
import useUserStore from "../store";

import type { MenuItem } from "@dzangolab/vue3-layout";

defineProps({
  noLocaleSwitcher: Boolean,
  showUserMenu: {
    default: true,
    required: false,
    type: Boolean,
  },
});

const userStore = useUserStore();

const { user } = storeToRefs(userStore);

const { layout: layoutConfig, user: userConfig } = useConfig();

const home = computed(() => {
  return user.value
    ? userConfig && userConfig?.routes?.home
      ? userConfig.routes.home
      : "profile"
    : layoutConfig && layoutConfig?.homeRoute
      ? layoutConfig.homeRoute
      : undefined;
});

const router = useRouter();

const allRoutes = router.getRoutes();

const routes = computed(() => {
  if (!user.value) {
    return layoutConfig?.mainMenu?.filter((item) => {
      const route = allRoutes.find((r) => r.name === item.route);

      return route && !route.meta?.authenticated;
    }) as MenuItem[];
  }

  return layoutConfig?.mainMenu as MenuItem[];
});
</script>
