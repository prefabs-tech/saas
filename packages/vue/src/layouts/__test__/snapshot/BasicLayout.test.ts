import configPlugin from "@dzangolab/vue3-config";
import i18nPlugin from "@dzangolab/vue3-i18n";
import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import { createPinia } from "pinia";
import { describe, expect, it } from "vitest";
import { createRouter, createWebHistory } from "vue-router";

import BasicLayout from "../../BasicLayout.vue";
import appConfig from "../config";

import type { VueWrapper } from "@vue/test-utils";

describe("BasicLayout", () => {
  it("matches snapshot", () => {
    const pinia = createPinia();
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/",
          name: "home",
          component: BasicLayout,
        },
      ],
    });

    const wrapper: VueWrapper = shallowMount(BasicLayout, {
      global: {
        plugins: [
          pinia,
          [
            configPlugin,
            {
              config: appConfig,
            },
          ],
          [
            i18nPlugin,
            {
              config: appConfig,
            },
          ],
          router,
        ],
        stubs: {
          RouterLink: RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
