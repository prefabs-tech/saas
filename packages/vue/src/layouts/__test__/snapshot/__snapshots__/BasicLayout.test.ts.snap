// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasicLayout > matches snapshot 1`] = `
<basic-layout-stub
  nolocaleswitcher="false"
/>
`;
